src/components/FormField/Select.tsx

```diff
@@ -0,0 +1,186 @@
+  width: 100%;
+`;
+const SelectTrigger = styled(RadixSelect.Trigger)<{ error: boolean }>`
+  width: 100%;
+  ${FormElement};
+  ${({ error, theme }) =>
+    error
+      ? `
+      border: 1px solid ${theme.click.field.color.stroke.error} !important;
+      background: ${theme.click.field.color.background.active} !important;
+       color:${theme.click.field.color.text.error} !important;`
+      : ""}
+`;
+const SelectContent = styled(RadixSelect.Content)`
+  ${MenuContent};
+  &[data-state="open"] ~ label,
+  &:focus ~ label {
+    ${({ theme }) => `
+      color: ${theme.click.field.color.label.active};
+      font: ${theme.click.field.typography.label.active};
+    `};
+  }
+`;
+
+const SelectViewport = styled(RadixSelect.Viewport)`
+  width: 100%;
+`;
+
+const Select = ({
+  placeholder = "Select an option",
+  label,
+  children,
+  disabled,
+  id,
+  error,
+  value,
+  defaultValue,
+  onValueChange,
+  open,
+  defaultOpen,
+  onOpenChange,
+  dir,
+  name,
+  required,
+  ...props
+}: Props) => {
+  id = id ?? uniqueId("select");
+  return (
+    <FormRoot {...props}>
+      {error && <Error>{error}</Error>}
+      <SelectRoot
+        value={value}
+        defaultValue={defaultValue}
+        onValueChange={onValueChange}
+        open={open}
+        defaultOpen={defaultOpen}
+        onOpenChange={onOpenChange}
+        dir={dir}
+        name={name}
+        required={required}
+        disabled={disabled}
+      >
+        <SelectTrigger id={id} error={typeof error !== "undefined"}>
+          <RadixSelect.Value placeholder={placeholder} />
+          <RadixSelect.Icon>
+            <Icon name='sort' />
+          </RadixSelect.Icon>
+        </SelectTrigger>
+        <RadixSelect.Portal>
+          <SelectContent position='popper' sideOffset={5}>
+            <RadixSelect.ScrollUpButton>
+              <ScrollButton>
+                <Icon name='chevron-up' />
+              </ScrollButton>
+            </RadixSelect.ScrollUpButton>
+            <SelectViewport>{children}</SelectViewport>
+            <RadixSelect.ScrollDownButton>
+              <ScrollButton>
+                <Icon name='chevron-down' />
+              </ScrollButton>
+            </RadixSelect.ScrollDownButton>
+          </SelectContent>
+        </RadixSelect.Portal>
+      </SelectRoot>
+      {label && (
+        <Label
+          htmlFor={id}
+          disabled={disabled}
+          error={typeof error !== "undefined"}
+        >
+          {label}
+        </Label>
+      )}
+    </FormRoot>
+  );
+};
+interface GroupProps extends RadixSelect.SelectGroupProps {
+  label?: ReactNode;
+}
+
+const SelectGroup = styled(RadixSelect.Group)`
+  ${OptionContainer};
+  flex-direction: column;
+  align-items: flex-start;
+  padding: 0;
+  justify-content: center;
+`;
+
+const SelectGroupLabel = styled(RadixSelect.Label)`
+  font-family: Inter;
+  font-size: 0.75rem;
+  font-style: normal;
+  font-weight: 500;
+  line-height: 150%;
```

Comment:
this should probably be a token

Reply:
Asked for the token will update on receiving it